CONTRIBUTING.md

```diff
@@ -0,0 +1,145 @@
+- `duplicate` - A duplicate signifies that another issue encapsulates or restates this issue. Duplicates will be closed and referenced as such. 
+- `ehancement` - An ehancement is a feature request. Please provide as much detail as possible and consider whether your idea fits the scope and aims of this project.
+- `help wanted` - Issues we need or would love help from the community to resolve.
+- `pending response` - This label indicates that the issue is awaiting reply from a contributor or a maintainer.
+- `question` - Use the question label to indicate you are having a problem with something and would like feedback from the community.
+- `wontfix` - The issue is legitimate, but it is not something the team is currently able or willing to fix or implement. Issues with this label may be revisited in the future.
+
+## React Component Contribution Process
+
+React component contributions should observe the following high level process:
+
+1. Check the [Patternfly Decision Tree](https://github.com/patternfly/patternfly-design/blob/master/resources/decision-tree/PatternflyDecisionTree.pdf) to see if your component meets the criteria for an addition to Patternfly.
+2. Create an `enhancement` issue to indicate you'd like to contribute this component to Patternfly React and notify the Patternfly React maintainer team.
+3. Implement the component observing our [react component requirements](#react-component-requirements) 
+and [code contribution guidelines](#code-contribution-guidelines)
+4. Submit your pull request and be sure to reference the `enhancement` issue so that others interested in this component may discuss. 
+5. Test your component. Once the pull request is merged, please test the component can successfully be used as intended downstream.
+
+## React Component Requirements 
+
+Please ensure that all React UI components contributed meet the following guidelines:
+
+* This repository serves as a UI / presentational component library only. This means we should not be introducing container components which subscribe to state updates or handle data fetching (i.e. redux aware components). Prefer [stateless functional components](http://buildwithreact.com/article/stateless-functional-components)
+when possible and accept [props](https://facebook.github.io/react/docs/components-and-props.html) as UI display parameters.
+* Provide a [single default export](http://exploringjs.com/es6/ch_modules.html#_single-default-export) for exporting your React UI component as an ES6 Module in your component's jsx definition.
+* Provide an associated `.stories.js` [story](https://getstorybook.io/docs/react-storybook/basics/writing-stories) for your component. Stories should demonstrate as many different UI states for your component as possible. Use Storybook [knobs](https://github.com/storybooks/storybook-addon-knobs) to enable dynamic visualizations of your component's props.
```

Comment:
Should we also include comments about the template that we have defined, such as: 

> Include a link to the design documentation in the [PatternFly Pattern Library[(http://www.patternfly.org/pattern-library/) if it exists. If a PatternFly design does not exist yet, then provide a description that explains when the component would be used and what goal or task it helps to accomplish.

If we do add this info, then maybe we should move some of the storybook details to sub-bullets?